container/client.go

```diff
@@ -37,31 +37,41 @@ type Client interface {
 //  * DOCKER_HOST			the docker-engine host to send api requests to
 //  * DOCKER_TLS_VERIFY		whether to verify tls certificates
 //  * DOCKER_API_VERSION	the minimum docker api version to work with
-func NewClient(pullImages, enableLabel bool) Client {
+func NewClient(pullImages, enableLabel bool, checkAll bool) Client {
 	cli, err := dockerclient.NewEnvClient()
 
 	if err != nil {
 		log.Fatalf("Error instantiating Docker client: %s", err)
 	}
 
-	return dockerClient{api: cli, pullImages: pullImages, enableLabel: enableLabel}
+	return dockerClient{
+		api:         cli,
+		pullImages:  pullImages,
+		enableLabel: enableLabel,
+		checkAll:    checkAll}
 }
 
 type dockerClient struct {
 	api         *dockerclient.Client
 	pullImages  bool
 	enableLabel bool
+	checkAll    bool
 }
 
 func (client dockerClient) ListContainers(fn Filter) ([]Container, error) {
 	cs := []Container{}
 	bg := context.Background()
 
-	log.Debug("Retrieving running containers")
+	if client.checkAll {
+		log.Debug("Retrieving containers")
+	} else {
+		log.Debug("Retrieving running containers")
+	}
 
 	runningContainers, err := client.api.ContainerList(
 		bg,
-		types.ContainerListOptions{})
+		types.ContainerListOptions{
+			All: client.checkAll})
```

Comment:
By setting `All` we would also include `restarting` and `removing` containers, which might have some undefined behaviour if watchtower tries to update them. It would be more robust to add a filter with specific states: `running`, `exited` .